addons/sale/models/sale.py

```diff
@@ -625,6 +625,13 @@ def _create_invoices(self, grouped=False, final=False):
         else:
             out_invoice_vals_list = invoice_vals_list
 
+        if invoice_vals['type'] in self.env['account.move'].get_outbound_types():
+            invoice_bank_id = self.partner_id.bank_ids and self.partner_id.bank_ids[0]
```

Comment:
`invoice_bank_id = self.partner_id.bank_ids[:1]` gives the same result and is safe

Reply:
updated, thanks for the hint